### PR TITLE
Port telemetry to FeatureStore API

### PR DESCRIPTION
**Pull request by jklegar**
Signed-off-by: Jacob Klegar <jacob@tecton.ai>

**What this PR does / why we need it**: Ports opt-out telemetry to the new FeatureStore API

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Opt-out telemetry (usage logging) is now enabled on the FeatureStore API. To disable it, see https://docs.feast.dev/v/master/feast-on-kubernetes/advanced-1/telemetry
```

### CODE REVIEW

**Comment by woop** [comment]
@jklegar would it make sense to add telemetry to all of our methods on the FeatureStore class?

File: sdk/python/feast/telemetry.py [hunk 1]
```diff
@@ -14,32 +14,71 @@
...
-    }
-    try:
-        requests.post(TELEMETRY_ENDPOINT, json=json)
-    except Exception:
-        pass
-    return
+try:
+    sdk_version = pkg_resources.get_distribution("feast").version
+except pkg_resources.DistributionNotFound:
+    sdk_version = "local build"
```
Comment (jklegar) [inline sdk/python/feast/telemetry.py:34]: I think this was accidentally left in fyi

**Review by woop — APPROVED** [review]

**Pull request merged**